src/compiler/sourcemap.ts

```diff
@@ -263,8 +302,17 @@ namespace ts {
             exit();
         }
 
+        function serializeMappings(len: number): string {
+            let mappings = "";
+            for (let i = 0; i < len; i += 1024) {
+                mappings += String.fromCharCode.apply(undefined, mappingsBuffer.subarray(i, Math.min(len, i + 1024)));
+            }
+            return mappings;
+        }
+
         function toJSON(): RawSourceMap {
             commitPendingMapping();
+            const mappings = (lastMappings ??= serializeMappings(mappingsPos));

```

Comment:
This seems sketchy. Do we guarantee that no new mappings will ever be added after calling `toJSON()`?